agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -377,6 +387,11 @@ func (state *DockerTaskEngineState) RemoveTask(task *apitask.Task) {
 
 	for _, dockerContainer := range containerMap {
 		state.removeIDToContainerTaskUnsafe(dockerContainer)
+
+		v3EndpointID := dockerContainer.Container.V3EndpointID
```

Comment:
Can you add a getter in Container instead of referencing it directly? And you can combine this into the next line:
```
if dockerContainer.Container.GetV3EndpointID() == ""{
}
```